### ISSUE TITLE
Remove the ResourceModelCache SPI

### ISSUE THREAD

**Issue by olaf-otto** [unic/neba#338]
NEBA currently features an extension point for providing custom application-level caching for resource models, the ResourceModelCache. This cache, in theory, allows users to extend the abilities of NEBA's built-in request-scoped resource model cache, e.g. by introducing application-scoped model caching.

However, doing so is risky (memory leaks, model dependencies, classloading issues when caching models beyond the lifetime of their source bundles....). Also is likely to yield little benefit compared to established means of caching using HTTP-caching (Dispatcher) or Application-Level HTTP-Caching (https://adobe-consulting-services.github.io/acs-aem-commons/features/http-cache/index.html).

Furthermore, the possible alteration of the scope of a resource model has further ramnifications: Any metadata related to a model instance - such as what specific resource was looked up, what resource type was looked  - must be kept in the same scope for consistency (such metadata is necessary e.g. for #335). However, the resource model cache can only cache models and not internal metadata, making it impossible to guarantee this consistency.

In summary, it makes sense to remove this SPI as it introduces strong architectural issues for little to no value.

**Comment by olaf-otto** [comment]
Done
